### DISCUSSION TITLE
How to use context of NavigationMenu?

### DISCUSSION THREAD

**Discussion by 7iomka** [unovue/reka-ui#1274] [Q&A]
Hi.
It is possible somehow to share 
conext of NavigationMenu defined [here](https://github.com/radix-vue/radix-vue/blob/main/packages/radix-vue/src/NavigationMenu/NavigationMenuRoot.vue#L74C15-L74C42)?
I used shadcn-vue
It is a wrapper around  radix-vue.
So, I want [inside List wrapper](https://github.com/radix-vue/shadcn-vue/blob/dev/apps/www/src/lib/registry/default/ui/navigation-menu/NavigationMenuList.vue) to have ability to access to `direction` prop, passed for Root.
Using injection of context is impossible, because it's not exported from radix-vue.
So, any workaround for this situation?

**Comment by Dodobibi** [comment]
It would be nice if radix-vue export `inject***Context` for all components.
Is there a reason for not do it ?

**Reply by zernonia** [reply]
It actually make sense to expose `injectXXXContext`. We wanna enable dev to construct as powerful component as possible and not limited to the internal logic. 

We can create a feature ticket for this, however it will be planned for v2 as it's almost ready 😁

**Reply by Dodobibi** [reply]
As these are simple (but many) exports, would you like a PR now?

**Reply by zernonia** [reply]
Cause there will be some changes in terms of component/component name, and I wouldn't want to expose it at this point as it would inconvenient the migration steps more..

**Reply by zernonia** [reply]
Created the issue https://github.com/radix-vue/radix-vue/issues/1281
